projects/or_scs/src/Mirza/OrgRegistry/Handlers/Keys.hs

```diff
@@ -206,52 +211,42 @@ keyStateQuery kid = do
 -- or revoked. The function can be modified in the future to add additional
 -- constraints that must be checked before the key is updated in anyway
 -- (effectively controling the minimum state for write access to the key).
-protectKeyUpdate :: ( Member err     '[AsORKeyError])
-                 => CT.ORKeyId
-                 -> CT.UserId
+protectKeyUpdate :: ( Member err     '[AsORKeyError, AsORError])
+                 => AuthUser
+                 -> CT.ORKeyId
                  -> DB context err ()
-protectKeyUpdate keyId userId = do
+protectKeyUpdate authUser keyId = do
   -- Although this check is implictly performed in keyStateQuery we explicitly perform it here so that the code reads
   -- logically and progressively builds constraints in order and so that the correct error message is shown when the
   -- keyId is not found, rather then failing because the user id check in doesUserOwnKeyQuery fails.
-  key <- getKeyById keyId
-  when (isNothing key) $ throwing _KeyNotFoundORKE keyId
+  maybeKey <- getKeyById keyId
 
-  userOwnsKey <- doesUserOwnKeyQuery userId keyId
-  unless userOwnsKey $ throwing_ _UnauthorisedKeyAccessORKE
+  key <- maybe (throwing _KeyNotFoundORKE keyId) pure maybeKey
+  _ <- userOrganisationAuthorisationQuery authUser $ orgPrimaryKeyToGS1CompanyPrefix $ key_org key
```

Comment:
Is this being done just to check that an error is not being thrown?

Reply:
Yeah basically. Specifically the intention of that function is to throw when any of the constraints are violated.